libraries/AP_GPS/AP_GPS_SBF.cpp

```diff
@@ -415,7 +418,21 @@ AP_GPS_SBF::process_message(void)
         if (temp.Latitude > -200000) {
             state.location.lat = (int32_t)(temp.Latitude * RAD_TO_DEG_DOUBLE * (double)1e7);
             state.location.lng = (int32_t)(temp.Longitude * RAD_TO_DEG_DOUBLE * (double)1e7);
-            state.location.alt = (int32_t)(((float)temp.Height - temp.Undulation) * 1e2f);
+
+            state.have_height_above_WGS84 = true;
+            state.height_above_WGS84 = float(temp.Height);
+
+            // CHECK THIS likely should be set to never return ASML........ terrain following issues?.....
+            // See "mosaic-X5 Reference Guide": page 137 Section 3.2.9 Datum Definition 
+            state.location.alt = gps.get_location_altitude_frame(int32_t((float(temp.Height) - temp.Undulation)* 100), int32_t(temp.Height * 100));
+
+            // DEBUG to test the GPS
```

Comment:
Note: this is here to allow someone who has a Septentrio GPS to test if the undulation is ever returned with respect to EGM96 AKA AMSL as used in ArduPilot. From my reading of the mosaic-H  or X5 datasheets that geoid is not available

"The geoid undulation specifies the local difference between the geoid and the WGS84 ellipsoid."
 That is normal. but the Default Geoid is WGS84 itself.
 
![image](https://user-images.githubusercontent.com/69225461/183470982-20ebf99b-ee4d-44f0-ab30-869de8b09d4b.png)

Reply:
I have a unit, I can try and cross check that for you. For the record the exsisting code here was needed to ensure consistency with what we were producing from a u-blox unit. Without it I was seeing 10-20 meter discrepancies.